src/level/level.cpp

```diff
@@ -10,35 +10,51 @@
 
 
 namespace level {
-    Level::Level(Size size) {
-        //TODO(ang): mettere il giocatore davanti alla porta, o su punto di spawn prestabilito
-        // (quando la stanza sarà nuova last player position conterrà il valore dell'entrata)
-        lastPlayerPosition_ = Position(1,1); 
-
+    Level::Level(loader::LoaderHandler *loader) {
         segment_ = datastruct::Vector<DisplayableSegment *>();
-        // creazione dei muri esterni
-        // TODO(ang): creare una funzione per creare i muri esterni
-        // questo è temporaneo
-        using namespace enums; // Direction::RIGHT, Direction::LEFT, Direction::UP, Direction::DOWN
-        segment_.push_back((DisplayableSegment *) new WallSegment(
-            Position(0, 0), Direction::RIGHT, size.colonna - 1));
-        segment_.push_back((DisplayableSegment *) new DoorSegment(
-            Position(0, size.colonna - 1), Direction::DOWN, 5));
-        segment_.push_back((DisplayableSegment *) new WallSegment(
-            Position(5, size.colonna - 1), Direction::DOWN, size.riga - 6));
-        segment_.push_back((DisplayableSegment *) new WallSegment(
-            Position(size.riga - 1, size.colonna - 1), Direction::LEFT, size.colonna - 1));
-        segment_.push_back((DisplayableSegment *) new WallSegment(
-            Position(size.riga - 1, 0), Direction::UP, size.riga - 1));
-        // TODO(ang): gestire anche la creazione dei segmenti intermedi 
+
+        datastruct::Vector<WallSegment *> *segments = nullptr;
+        segments = loader->wallLoader->getLoadedObjects(); 
+        if (segments != nullptr) {
+            for (unsigned int i = 0; i < segments->size(); i++) {
+                segment_.push_back((DisplayableSegment *) segments->at(i));
+            }
+            delete segments;
+        }
+
+        datastruct::Vector<DoorSegment *> *doors = nullptr;
+        doors = loader->doorLoader->getLoadedObjects();
+        if (doors != nullptr) {
+            numOfDoors_ = doors->size();
+            for (unsigned int i = 0; i < doors->size(); i++) {
+                segment_.push_back((DisplayableSegment *) doors->at(i));
```

Comment:
non mi piace il fatto che fai la delete dell'oggetto qui, non puoi semplicemente trasferire il vettore?

Reply:
boh fa il suo dovere però non mi piace che faccia una deep copy per poi eliminare il vecchio, a questo punto fai prima a tenerteli tutto in memoria, già loadati, fai la deep copy di uno a caso e poi non elimini

Reply:
anche perchè così se ha un errore a caso l'ha prima che incominci non mentre giochi

Reply:
Il problema di tenerli sempre in memoria sin da subito è che sono pointers, puoi utilizzarli sono una volta.  
Cioè se vuoi riutilizzare il template di un livello, devi comunque duplicarlo (quindi allocare di nuovo memoria per nuovi oggetti), ergo farci una deep copy dell'array, quindi alla fine il problema c'è comunque. 

Inoltre non è una deep copy tanto dispendiosa, sono copie di pointers di oggetti, non di oggetti stessi, e poi un numero costante, dato che non credo esisteranno mai (anche per la nostra sanità mentale) template che vanno sulle 100+ segmenti.

Reply:
Il punto è che così non deve leggere sempre dai file

Reply:
Comunque è uguale